sdk/identity/identity/src/credentials/managedIdentityCredential.ts

```diff
@@ -337,15 +337,25 @@ export class ManagedIdentityCredential implements TokenCredential {
       } else {
         throw new CredentialUnavailable("The managed identity endpoint is not currently available");
       }
-
       return result;
     } catch (err) {
       span.setStatus({
         code: CanonicalCode.UNKNOWN,
         message: err.message
       });
-      throw err;
+
+      if (err.code == "ENETUNREACH") {
+        throw new CredentialUnavailable("ManagedIdentityCredential is unavailable. No managed identity endpoint found.");
+      }
```

Comment:
Should condition also include err being from a request which fails with a status code of 400 or is this handled above?

Reply:
Discussed offline, since fixing this functionaliy is a bit out of the scope of this PR we will file a separate issue to fix.